services/navigation/builtin/builtin.go

```diff
@@ -121,14 +117,17 @@ type builtIn struct {
 	metersPerSec            float64
 	degPerSec               float64
 	logger                  golog.Logger
-	cancelCtx               context.Context
 	cancelFunc              func()
+	navOnceCancelFunc       func()
```

Comment:
Can we name the cancel functions more descriptively? Maybe `wholeServiceCancelFunc` and `currentWaypointCancelFunc`?